app/models/course/discussion/topic.rb

```diff
@@ -19,7 +19,9 @@ def subscribed_by?(user)
   # @param [User] user The user who needs to subscribe to this topic
   def ensure_subscribed_by(user)
     return true if self.subscribed_by?(user)
-    subscriptions.build(user: user).save!
+    ActiveRecord::Base.transaction do
+      subscriptions.build(user: user).save!
```

Comment:
OK, So even we have the `rescue ActiveRecord::RecordInvalid`, the enclosing transaction will still rollback ?